src/ansys/units/dimensions.py

```diff
@@ -62,41 +64,40 @@ def _temp_precheck(self, dims2, op: str = None) -> Optional[str]:
             if (dims1 == temp and dims2 == temp) and op == "-":
                 return ansunits.Dimensions(dimensions_container=delta_temp)
 
-    @property
-    def dimensions(self):
-        """A dictionary representation."""
-        return self._dimensions
-
     def __str__(self):
-        dims = {x.name: y for x, y in self._dimensions.items()}
+        dims = {x.name: y for x, y in self}
```

Comment:
I suppose if `__str__` and `__repr__` are identical you can encapsulate the code in a private function and test for this condition in `test_dimensions`.  i.e.

```
assert str(dim) == repr(dim)
```

somewhere in the unit tests.